strict.test_atexit_standalone regression: "undefined symbol: __syscall_pause"

After PR #26860

```
commit 4bf5fb7a35c38935d20c6f8a7e3a3f1153932758
Author: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date:   Thu May 21 22:24:38 2026 +0200

    Update musl to v1.2.6 (#26860)
```

test `strict.test_atexit_standalone` fails with

```

C:\emsdk\emscripten\main>test\runner strict.test_atexit_standalone
Running 1 tests
[1/1] test_atexit_standalone (test_core.strict.test_atexit_standalone) ... FAIL

Stdout:
wasm-ld: error: C:\emsdk\emscripten\main\cache\sysroot\lib\wasm32-emscripten\libstandalonewasm-nocatch-pure.a(exit.o): undefined symbol: __syscall_pause
emcc: error: ''C:/emsdk/llvm/git/build_main_vs2022_64/Release/bin\wasm-ld.exe' -o test_atexit.wasm 'C:\Users\clb\AppData\Local\Temp\tmp1dpzghcilibemscripten_js_symbols.so' --strip-debug --export-if-defined=__start_em_asm --export-if-defined=__stop_em_asm --export-if-defined=__start_em_lib_deps --export-if-defined=__stop_em_lib_deps --export-if-defined=__start_em_js --export-if-defined=__stop_em_js --export-table -z stack-size=65536 --no-growable-memory --initial-heap=16777216 --stack-first --table-base=1 'C:\Users\clb\AppData\Local\Temp\emscripten_temp_3q07pvdp\test_atexit.o' '-LC:\emsdk\emscripten\main\cache\sysroot\lib\wasm32-emscripten' '-LC:\emsdk\emscripten\main\src\lib' 'C:\emsdk\emscripten\main\cache\sysroot\lib\wasm32-emscripten\crt1.o' -lstandalonewasm-nocatch-pure -lc-debug -ldlmalloc-debug -lcompiler_rt -lsockets --fatal-warnings -mllvm -combiner-global-alias-analysis=false -mllvm -disable-lsr' failed (returned 1)

None


======================================================================
FAIL: test_atexit_standalone (test_core.strict.test_atexit_standalone)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\common.py", line 1213, in run_process
    rtn = utils.run_process(cmd, check=check, **kwargs)
  File "C:\emsdk\emscripten\main\tools\utils.py", line 45, in run_process
    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
  File "C:\emsdk\python\3.13.3_64bit\Lib\subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\emsdk\\emscripten\\main\\emcc.bat', 'C:\\emsdk\\emscripten\\main\\test\\core\\test_atexit.c', '-o', 'test_atexit.js', '-sNO_DEFAULT_TO_CXX', '-sSTRICT', '-sSTANDALONE_WASM', '-sPURE_WASI', '-Wclosure', '-Werror', '-Wno-limited-postlink-optimizations', '-Wno-unused-command-line-argument', '-Wno-unused-command-line-argument']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\common.py", line 380, in resulting_test
    return func(self, *args)
  File "C:\emsdk\emscripten\main\test\decorators.py", line 496, in metafunc
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 2540, in test_atexit
    self.do_core_test('test_atexit.c')
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\test_core.py", line 501, in do_core_test
    self.do_run_in_out_file_test(Path('core', testname), **kwargs)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1404, in do_run_in_out_file_test
    output = self._build_and_run(srcfile, expected, **kwargs)
  File "C:\emsdk\emscripten\main\test\common.py", line 1423, in _build_and_run
    js_file = self.build(filename, **kwargs)
  File "C:\emsdk\emscripten\main\test\common.py", line 913, in build
    self.run_process(cmd, stderr=self.stderr_redirect if not DEBUG else None)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1218, in run_process
    self.fail(f'subprocess exited with non-zero return code({e.returncode}): `{shlex.join(cmd)}`')
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: subprocess exited with non-zero return code(1): `'C:\emsdk\emscripten\main\emcc.bat' 'C:\emsdk\emscripten\main\test\core\test_atexit.c' -o test_atexit.js -sNO_DEFAULT_TO_CXX -sSTRICT -sSTANDALONE_WASM -sPURE_WASI -Wclosure -Werror -Wno-limited-postlink-optimizations -Wno-unused-command-line-argument -Wno-unused-command-line-argument`

Stdout:
wasm-ld: error: C:\emsdk\emscripten\main\cache\sysroot\lib\wasm32-emscripten\libstandalonewasm-nocatch-pure.a(exit.o): undefined symbol: __syscall_pause
emcc: error: ''C:/emsdk/llvm/git/build_main_vs2022_64/Release/bin\wasm-ld.exe' -o test_atexit.wasm 'C:\Users\clb\AppData\Local\Temp\tmp1dpzghcilibemscripten_js_symbols.so' --strip-debug --export-if-defined=__start_em_asm --export-if-defined=__stop_em_asm --export-if-defined=__start_em_lib_deps --export-if-defined=__stop_em_lib_deps --export-if-defined=__start_em_js --export-if-defined=__stop_em_js --export-table -z stack-size=65536 --no-growable-memory --initial-heap=16777216 --stack-first --table-base=1 'C:\Users\clb\AppData\Local\Temp\emscripten_temp_3q07pvdp\test_atexit.o' '-LC:\emsdk\emscripten\main\cache\sysroot\lib\wasm32-emscripten' '-LC:\emsdk\emscripten\main\src\lib' 'C:\emsdk\emscripten\main\cache\sysroot\lib\wasm32-emscripten\crt1.o' -lstandalonewasm-nocatch-pure -lc-debug -ldlmalloc-debug -lcompiler_rt -lsockets --fatal-warnings -mllvm -combiner-global-alias-analysis=false -mllvm -disable-lsr' failed (returned 1)

None

----------------------------------------------------------------------
Ran 1 test in 0.472s

FAILED (failures=1)

C:\emsdk\emscripten\main>
```

--- a/system/lib/standalone/standalone.c
+++ b/system/lib/standalone/standalone.c
@@ -115,6 +115,10 @@ weak int __syscall_newfstatat(int dirfd, intptr_t path, intptr_t buf, int flags)
   return -ENOSYS;
 }
 
+weak int __syscall_pause() {
+  return -ENOSYS;
+}
+
 weak int __syscall_lstat64(intptr_t path, intptr_t buf) {
   return -ENOSYS;
 }

--- a/test/codesize/test_codesize_mem_O3_grow_standalone.json
+++ b/test/codesize/test_codesize_mem_O3_grow_standalone.json
@@ -2,9 +2,9 @@
   "a.out.js": 4127,
   "a.out.js.gz": 1994,
   "a.out.nodebug.wasm": 5641,
-  "a.out.nodebug.wasm.gz": 2660,
+  "a.out.nodebug.wasm.gz": 2659,
   "total": 9768,
-  "total_gz": 4654,
+  "total_gz": 4653,
   "sent": [
     "args_get",
     "args_sizes_get",

--- a/test/codesize/test_codesize_mem_O3_standalone.json
+++ b/test/codesize/test_codesize_mem_O3_standalone.json
@@ -2,9 +2,9 @@
   "a.out.js": 4060,
   "a.out.js.gz": 1956,
   "a.out.nodebug.wasm": 5565,
-  "a.out.nodebug.wasm.gz": 2600,
+  "a.out.nodebug.wasm.gz": 2598,
   "total": 9625,
-  "total_gz": 4556,
+  "total_gz": 4554,
   "sent": [
     "args_get",
     "args_sizes_get",

--- a/test/codesize/test_codesize_mem_O3_standalone_narg.json
+++ b/test/codesize/test_codesize_mem_O3_standalone_narg.json
@@ -2,9 +2,9 @@
   "a.out.js": 3595,
   "a.out.js.gz": 1706,
   "a.out.nodebug.wasm": 5354,
-  "a.out.nodebug.wasm.gz": 2443,
+  "a.out.nodebug.wasm.gz": 2442,
   "total": 8949,
-  "total_gz": 4149,
+  "total_gz": 4148,
   "sent": [
     "proc_exit"
   ],

